package/MDAnalysis/auxiliary/base.py

```diff
@@ -431,6 +434,74 @@ def read_ts(self, ts):
+            the arguments to make sure the input is treated properly. Then, 
+            the AuxReader(s) with appropriate :attr:`data_selector` are
+            associated with the `coord_parent` from which `add_auxiliary` was
+            called. """
+        if "auxname" in kwargs:
+            # This check is necessary for the tests in coordinates/base
+            # `test_reload_auxiliaries_from_description`
+            # because there, auxreaders are created from descriptions without
+            # giving explicit `aux_spec`s
+            aux_spec = {kwargs["auxname"]: None}
+
+        elif aux_spec is None or aux_spec == "*":
+            # Add all terms found in the file if aux_spec is None
+            aux_spec = {term: term for term in self.terms}
+
+        elif isinstance(aux_spec, str):
+            # This is to keep XVGReader functioning as-is, working with strings
+            # for what used to be `auxname`. Often, no `auxterms` are specified
+            # for XVG files. The below sets the data selector to None,
+            # the default for XVGReader
+            aux_spec = {aux_spec: None}
+
+        for auxname in aux_spec:
+            if auxname in coord_parent.aux_list:
+                raise ValueError(f"Auxiliary data with name {auxname} already "
+                                 "exists")
+            if " " in auxname:
+                warnings.warn(f"Auxiliary name '{auxname}' contains a space. "
+                              "Only dictionary style accession, not attribute "
+                              f"style accession of '{auxname}' will work.")
+            description = self.get_description()
+            # make sure kwargs are also used
+            description_kwargs = {**description, **kwargs}
+            # Make a copy of the auxreader for every attribute to add
+            # This is necessary so all attributes can be iterated over
+            aux = auxreader(**description_kwargs)
+            aux.auxname = auxname
+            aux.data_selector = aux_spec[auxname]
+            coord_parent._auxs[auxname] = aux
+            coord_parent.ts = aux.update_ts(coord_parent.ts)
+
+        aux_memory_usage = 0
+        # Check if testing, which needs lower memory limit
+        if "memory_limit" in kwargs:
+            memory_limit = kwargs["memory_limit"]
+        else:
+            memory_limit = 1e+09
+        for reader in coord_parent._auxs.values():
+            aux_memory_usage += reader._memory_usage()
+        if aux_memory_usage > memory_limit:
+            conv = 1000000000  # convert to GB
+            warnings.warn("AuxReader: memory usage warning! "
+                          f"Auxiliary data takes up {aux_memory_usage/conv} "
+                          f"GB of memory (Warning limit: {memory_limit/conv} "
+                          "GB).")
+
+    def _memory_usage(self):
+        raise NotImplementedError("BUG: Override _memory_usage() "
```

Comment:
You can (probably) test this by using `delattr`

```python
def test_memoryusage_notimpl():
   .... create reader 
    delattr(XDRReader, '_memory_usage')
   ...
```

Which should leave the base class implementation which will raise `NotImplemented`

Reply:
This was a bit problematic, trying to delete _memory_usage just led to AttributeErrors... I have now written a minimal test class without the implementation though and that works

Reply:
perfect!